lib/flowtypes.js

```diff
@@ -0,0 +1,268 @@
+/* @flow */
+'use strict';
+
+export type MessageResponse<T> = {
+    type: string;
+    message: T; // in general, can be anything
+};
+
+export type DefaultMessageResponse = MessageResponse<Object>;
+
+export type DeviceDescriptor = {
+    path: string;
+    session: ?string;
+    product: number;
+    serialNumber: number;
+    vendor: number;
+};
+
+export type Transport = {
+    enumerate: (wait?: boolean, previous?: ?Array<DeviceDescriptor>) => Promise<Array<DeviceDescriptor>>;
+    acquire: (i: DeviceDescriptor, checkPrevious?: boolean) => Promise<{session: string}>;
+    supportsSync: boolean;
+    release: (s: string) => Promise;
+    releaseSync: (s: string) => void;
+    configure: (s: string) => Promise;
+    call: (s: string, type: string, msg: Object) => Promise<DefaultMessageResponse>;
+    callSync: (s: string, type: string, msg: Object) => DefaultMessageResponse;
+    version: string;
+};
```

Comment:
OK. But I really want to keep all the possible message types in one extra file, since they just take too much space and make the code they are in unreadable.